src/clr-angular/data/datagrid/interfaces/filter.interface.ts

```diff
@@ -5,10 +5,14 @@
  */
 import { Observable } from 'rxjs';
 
-export interface ClrDatagridFilterInterface<T> {
+export interface ClrDatagridFilterInterface<T, S = any> {
   isActive(): boolean;
 
   accepts(item: T): boolean;
 
   changes: Observable<any>;
+
+  state?: S;
```

Comment:
`readonly state?: S;`, to allow for a getter only if they want. Specifying `readonly` in the interface still lets them have an actual property if they want to, we just won't write to it.

Reply:
Done